docs/kns_apps_tutorial.md

```diff
@@ -0,0 +1,347 @@
+                    }
+                ],
+                "templates_settings": {
+                    "global_retry": 20,
+                    "global_retry_sleep": 15
+                },
+                "templates": [
+                    {
+                        "name": "google-sample",
+                        "type": "kubectl",
+                        "path": "google-sample.yml"
+                    }
+                ]
+            }
+        }
+    },
+    "image": {
+        "type": "image/png",
+        "data": ""
+    }
+}
+```
+
+Lets examine this file step by steps. We will only focus on the commands part, everything else is the same than traditional Jarvice apps.
+
+```json
+    "commands": {
+        "Hello": {
+```
+
+* `commands` is a list, so you can add any app entry points as needed. This can be useful to create an all-in-one app image for example.
+* `hello` is the name of the current entry point for the app.
+
+```json
+        "Hello": {
+            "path": "/usr/bin/true",
+            "name": "Start nested http server",
+            "description": "Start a nested http server for testing and to say hello world.",
+            "parameters": {},
+```
+
+For KNS jobs, `path` and `parameters` are not used, but need to be present for the app to be usable by Jarvice upstream component.
+`path` should be set to `/usr/bin/true` and `parameters` left as an empty dict.
+
+`name` and `description` will display as before some details about this entry point, for users to be able to choose between multiple if exist.
+
+```json
+            "nestedkubernetes": {
+                "kubernetes_version": "",
+                "enable_gotty_shell": "true",
+                "ingress_type": "subdomain",
+                "targets": [
+                    {
+                        "service_name": "web",
+                        "service_port": 8080,
+                        "service_namespace": "default"
+                    }
+                ],
+                "templates_settings": {
+                    "global_retry": 20,
+                    "global_retry_sleep": 15
+                },
+                "templates": [
+                    {
+                        "name": "google-sample",
+                        "type": "kubectl",
+                        "path": "google-sample.yml"
+                    }
+                ]
+            }
+```
+
+`nestedkubernetes` is the new key that allows to define inside it all needed elements for a Kubernetes based app.
+Lets see each parts of this new dict.
+
+### 2.1. General parameters
+
+```json
+            "nestedkubernetes": {
+                "kubernetes_version": "",
+```
+
+* `kubernetes_version` allows to specify the Kubernetes version to be used for the K3S nested cluster. Format is `v1.X.Y` with `X` the major and `Y` the minor versions. For example: `v1.28.11`. See the compatibility matrix here: https://vcluster.com/docs/v0.19/deploying-vclusters/compat-matrix
```

Comment:
First reference to vCluster. It is something that we have to ask the user to check? I think that as this is not obvious, it would be better to specify just the list of possible Kubernetes versions that are supported.

Reply:
Good point, I will fix that.